.eslintrc.cjs

```diff
@@ -55,6 +55,21 @@ module.exports = {
     // so that floating-promises can be explicitly permitted with void operator
     'no-void': ['error', { allowAsStatement: true }],
 
+    // The rule is “safe await separator" which implements the architectural
+    // goal of “clearly separate an async function's synchronous prelude from
+    // the part that runs in a future turn”. That is our architectural rule. It
+    // can be trivially satisfied by inserting a non-nested `await null` in an
+    // appropriate place to ensure the rest of the async function runs in a
+    // future turn.  “sometimes synchronous" is a bug farm for particularly
+    // pernicious bugs in which you can combine two correct pieces of code to
+    // have emergent incorrect behavior.  It’s absolutely critical for shared
+    // service code. That means contracts, but it also means kernel components
+    // that are used by multiple clients. So we enable it throughout the repo
+    // and aim for no exceptions.
+    //
+    // TODO the default is 'warn', but upgrade this to 'error' when possible
```

Comment:
Not a blocker but mind making a ticket for this? Then it can be referenced here.

Reply:
Actually, I upgraded to error and fixed the remaining problems.